### DISCUSSION TITLE
Как отправлять ack/nack?

### DISCUSSION THREAD

**Discussion by ghost** [Lancetnik/Propan#135] [Questions]
> В большенстве случаев **Propan** сам совершает все необходимые действия: однако, если вы хотите управлять жизенным циклом сообщений напрямую, вы можете получить доступ к самому объекту сообщения и вызывать соотвествующие методы напрямую. 

Документация куцая, не нашел ничего про отправку ack/nack. Подскажите, как это делать?

**Comment by ghost** [comment]
гм.. после написания поста нашел https://lancetnik.github.io/Propan/getting_started/4_broker/2_routing/#error-handling

**Comment by Lancetnik** [comment]
@sergeykorablin вы можете получить доступ к оригинальному сообщению фреймворка, который используется под капотом через контекст, а у него уже дергать методы, которые вам нужны: в т.ч. ack/nack/reject и тд

```python
@broker.handle(...)
async def handler(msg_body, message = Context()):
    await message.ack()
```

Это общий вариант для любого брокера, но если нужны тайпхинты, используйте аннотацию для сообщения из `propan.annotations`

**Comment by Lancetnik** [comment]
Собственно да, вот про контекст https://lancetnik.github.io/Propan/ru/getting_started/5_dependency/2_context/
И вот про обработку ошибок https://lancetnik.github.io/Propan/ru/getting_started/4_broker/2_routing/

**Reply by ghost** [reply]
Спасибо!
